### ISSUE TITLE
Crash when sharing to Nextcloud and parent directory of last upload directory was renamed

### ISSUE THREAD

**Issue by cnmicha** [nextcloud/android#11495]
### ⚠️ Before posting ⚠️

- [X] This is a **bug**, not a question or an enhancement.
- [X] I've [searched for similar issues](https://github.com/nextcloud/android/issues) and didn't find a duplicate.
- [X] I've written a clear and descriptive title for this issue, not just "Bug" or "Crash".
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Steps to reproduce

1. Create a directory /A
2. Create a directory /A/B
3. Share a file to nextcloud and upload it to /A/B
4. Rename the directory /A to /C, so that B is now at /C/B
5. Share a file to Nextcloud
6. The locally cached content of /A/B is shown
7. Click the back icon on the left side of the top toolbar to switch to the parent directory

### Expected behaviour

It switches to /C/B or at least to the root directory 

### Actual behaviour

The Nextcloud app crashes, probably because the parent directory of /A/B, which is /A, doesn't exist anymore (since it was renamed to /C).
This makes the share-to-Nextcloud feature completely unusable until I clear the app data and setup everything on the Android client side from scratch.

### Android version

13

### Device brand and model

Samsung Galaxy A53

### Stock or custom OS?

Stock

### Nextcloud android app version

3.24.2

### Nextcloud server version

25.0.6

### Using a reverse proxy?

Yes

### Cause of error
```java
Exception in thread "Thread-13" java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String com.owncloud.android.datamodel.OCFile.getRemotePath()' on a null object reference
    at com.owncloud.android.operations.RefreshFolderOperation.run(RefreshFolderOperation.java:230)
    at com.owncloud.android.lib.common.operations.RemoteOperation.run(RemoteOperation.java:399)
    at java.lang.Thread.run(Thread.java:1012)

```

### App information
* ID: `com.nextcloud.client`
* Version: `30240290`
* Build flavor: `gplay`

### Device information
* Brand: `samsung`
* Device: `a53x`
* Model: `SM-A536B`
* Id: `TP1A.220624.014`
* Product: `a53xnaeea`

### Firmware
* SDK: `33`
* Release: `13`
* Incremental: `A536BXXS5CWB6`

**Comment by Ostlaender** [comment]
### Cause of error
```java
Exception in thread "Thread-6" android.app.ForegroundServiceStartNotAllowedException: startForegroundService() not allowed due to mAllowStartForeground false: service com.t_systems.android.webdav/com.owncloud.android.files.services.FileUploader
    at android.app.ForegroundServiceStartNotAllowedException$1.createFromParcel(ForegroundServiceStartNotAllowedException.java:54)
    at android.app.ForegroundServiceStartNotAllowedException$1.createFromParcel(ForegroundServiceStartNotAllowedException.java:50)
    at android.os.Parcel.readParcelable(Parcel.java:3346)
    at android.os.Parcel.createExceptionOrNull(Parcel.java:2433)
    at android.os.Parcel.createException(Parcel.java:2422)
    at android.os.Parcel.readException(Parcel.java:2405)
    at android.os.Parcel.readException(Parcel.java:2347)
    at android.app.IActivityManager$Stub$Proxy.startService(IActivityManager.java:6969)
    at android.app.ContextImpl.startServiceCommon(ContextImpl.java:1927)
    at android.app.ContextImpl.startForegroundService(ContextImpl.java:1893)
    at android.content.ContextWrapper.startForegroundService(ContextWrapper.java:798)
    at com.owncloud.android.files.services.FileUploader.retryUpload(FileUploader.java:1075)
    at com.owncloud.android.files.services.FileUploader.retryFailedUploads(FileUploader.java:1118)
    at com.owncloud.android.utils.FilesSyncHelper.lambda$restartJobsIfNeeded$0(FilesSyncHelper.java:240)
    at com.owncloud.android.utils.FilesSyncHelper$$ExternalSyntheticLambda0.run(Unknown Source:10)
    at java.lang.Thread.run(Thread.java:920)

```

### App information
* ID: `com.t_systems.android.webdav`
* Version: `7219 (build #9243)`
* Build flavor: `gplay`

### Device information
* Brand: `samsung`
* Device: `beyond1`
* Model: `SM-G973F`
* Id: `SP1A.210812.016`
* Product: `beyond1lteeea`

### Firmware
* SDK: `31`
* Release: `12`
* Incremental: `G973FXXSGHWC2`

**Comment by Ostlaender** [comment]
### Cause of error
```java
Exception in thread "Thread-6" android.app.ForegroundServiceStartNotAllowedException: startForegroundService() not allowed due to mAllowStartForeground false: service com.t_systems.android.webdav/com.owncloud.android.files.services.FileUploader
    at android.app.ForegroundServiceStartNotAllowedException$1.createFromParcel(ForegroundServiceStartNotAllowedException.java:54)
    at android.app.ForegroundServiceStartNotAllowedException$1.createFromParcel(ForegroundServiceStartNotAllowedException.java:50)
    at android.os.Parcel.readParcelable(Parcel.java:3346)
    at android.os.Parcel.createExceptionOrNull(Parcel.java:2433)
    at android.os.Parcel.createException(Parcel.java:2422)
    at android.os.Parcel.readException(Parcel.java:2405)
    at android.os.Parcel.readException(Parcel.java:2347)
    at android.app.IActivityManager$Stub$Proxy.startService(IActivityManager.java:6969)
    at android.app.ContextImpl.startServiceCommon(ContextImpl.java:1927)
    at android.app.ContextImpl.startForegroundService(ContextImpl.java:1893)
    at android.content.ContextWrapper.startForegroundService(ContextWrapper.java:798)
    at com.owncloud.android.files.services.FileUploader.retryUpload(FileUploader.java:1075)
    at com.owncloud.android.files.services.FileUploader.retryFailedUploads(FileUploader.java:1118)
    at com.owncloud.android.utils.FilesSyncHelper.lambda$restartJobsIfNeeded$0(FilesSyncHelper.java:240)
    at com.owncloud.android.utils.FilesSyncHelper$$ExternalSyntheticLambda0.run(Unknown Source:10)
    at java.lang.Thread.run(Thread.java:920)

```

### App information
* ID: `com.t_systems.android.webdav`
* Version: `7219 (build #9243)`
* Build flavor: `gplay`

### Device information
* Brand: `samsung`
* Device: `beyond1`
* Model: `SM-G973F`
* Id: `SP1A.210812.016`
* Product: `beyond1lteeea`

### Firmware
* SDK: `31`
* Release: `12`
* Incremental: `G973FXXSGHWC2`

**Comment by Ostlaender** [comment]
### Cause of error
```java
Exception in thread "Thread-6" android.app.ForegroundServiceStartNotAllowedException: startForegroundService() not allowed due to mAllowStartForeground false: service com.t_systems.android.webdav/com.owncloud.android.files.services.FileUploader
    at android.app.ForegroundServiceStartNotAllowedException$1.createFromParcel(ForegroundServiceStartNotAllowedException.java:54)
    at android.app.ForegroundServiceStartNotAllowedException$1.createFromParcel(ForegroundServiceStartNotAllowedException.java:50)
    at android.os.Parcel.readParcelable(Parcel.java:3346)
    at android.os.Parcel.createExceptionOrNull(Parcel.java:2433)
    at android.os.Parcel.createException(Parcel.java:2422)
    at android.os.Parcel.readException(Parcel.java:2405)
    at android.os.Parcel.readException(Parcel.java:2347)
    at android.app.IActivityManager$Stub$Proxy.startService(IActivityManager.java:6969)
    at android.app.ContextImpl.startServiceCommon(ContextImpl.java:1927)
    at android.app.ContextImpl.startForegroundService(ContextImpl.java:1893)
    at android.content.ContextWrapper.startForegroundService(ContextWrapper.java:798)
    at com.owncloud.android.files.services.FileUploader.retryUpload(FileUploader.java:1075)
    at com.owncloud.android.files.services.FileUploader.retryFailedUploads(FileUploader.java:1118)
    at com.owncloud.android.utils.FilesSyncHelper.lambda$restartJobsIfNeeded$0(FilesSyncHelper.java:240)
    at com.owncloud.android.utils.FilesSyncHelper$$ExternalSyntheticLambda0.run(Unknown Source:10)
    at java.lang.Thread.run(Thread.java:920)

```

### App information
* ID: `com.t_systems.android.webdav`
* Version: `7219 (build #9243)`
* Build flavor: `gplay`

### Device information
* Brand: `samsung`
* Device: `beyond1`
* Model: `SM-G973F`
* Id: `SP1A.210812.016`
* Product: `beyond1lteeea`

### Firmware
* SDK: `31`
* Release: `12`
* Incremental: `G973FXXSGHWC2`

**Comment by joshtrichards** [comment]
Dup of #11207 (at least the OP is). There's a fix pending.

**Comment by joshtrichards** [comment]
Fixed in #11410 pending the next release. 

@Ostlaender Your issue looks completely unrelated. Based on the error output it looks like you need to fix the permissions on the app. Specifically checking (outside the NC app) Android's *App Info* -> *All Permissions*. It appears you don't have `run foreground service` listed. This isn't a bug, but a configuration matter with your installation/OS. If you're unable to resolve on your own through searching, I suggest posting a query on the [Nextcloud Help Forum](https://help.nextcloud.com)
> `Exception in thread "Thread-6" android.app.ForegroundServiceStartNotAllowedException: startForegroundService() not allowed due to mAllowStartForeground false: service com.t_systems.android.webdav/com.owncloud.android.files.services.FileUploader`
